controls/raddiagram/performance/tips-tricks.md

```diff
@@ -0,0 +1,100 @@
+
+* __Shapes Width and Height__: To improve layout time, it is important the shapes ploted on the diagram to be with equal width and height. Layouting the diagram with different size of the shapes could lead to slower performance.
+
+* __Suspend Connection Update__: For better performance is better to suspend the connection update before calling the __Layout()__ method.
+
+#### __[XAML] Example 1: Suspend connection update__
+{{region raddiagram-performance-tips-tricks_0}}
+	this.xDiagram.Connections.ForEach(x => RadDiagramConnection.SetIsAutoUpdateSuppressed((RadDiagramConnection)x, true)); 
+	this.xDiagram.Layout(LayoutType.Tree, settings);
+	this.xDiagram.Connections.ForEach(x => RadDiagramConnection.SetIsAutoUpdateSuppressed((RadDiagramConnection)x, false));
+	this.xDiagram.Connections.ForEach(x => x.Update());
+{{endregion}}
+
+* __Simplify Shapes Template__: In a scenario with a large number of shapes you can try to simplify the template of the shapes as much as possible. This way you can increase the initial load time. You can check the [Create Custom Shape]({%slug raddiagram-howto-create-custom-shape%}) help article which desribes how you can create a custom shape. __Example 2__ demonstrate possible template. 
+
+#### __[XAML] Example 2: Simplify Shapes Template__
+{{region raddiagram-performance-tips-tricks_1}}
+	<Style TargetType="{x:Type telerik:RadDiagramShape}">
+		<Setter Property="Template">
+			<Setter.Value>
+				<ControlTemplate TargetType="telerik:RadDiagramShape">
+					<Grid Background="Orange">
+						<ContentPresenter Margin="{TemplateBinding Padding}"
+							  HorizontalAlignment="{TemplateBinding HorizontalContentAlignment}"
+							  VerticalAlignment="{TemplateBinding VerticalContentAlignment}" />
+					</Grid>
+				</ControlTemplate>
+			</Setter.Value>
+		</Setter>
+	</Style>
+{{endregion}}
+
+* __Routing__: The routing functionality of the diagram provides several algorithms __(OrgTreeRouter, AStarRouter, InflatedRectRouter)__ which can used to reduce the crossing of the connection. In some cases, one can be musch faster than others. You can test your scenario with all of them and used the one which works best for your case.
+       
+## Optimize Copy Paste 
+
+* __Disable Selection__: When copying and pasting hundreds of shapes could lead to performance issues. This is because thousands of visual elements are added in a Canvas (every shape has borders, resizing / rotation thumbs, many other visuals in its control template). What you can do to optimize the process is to disable the selection behavior of the paste shapes. In order to do that you can create a custom class which derives from RadDiagram control and override the __Paste()__ method. Then you can create a dummy boolean value which we are going to use to handle the selection of the control in its __PreviewSelectionChanged__ event handler.
```

Comment:
Is the starting "When" actually needed? Also, more appropriate in the last sentence is to tell that RadDiagram will use the dummy boolean value, rather than "we".